In [1]:
import numpy as np
from numpy import percentile
from numpy.random import rand

In [2]:
#generating sample data set of 1000 numbers
data=rand(1000)

In [3]:
#calculate quartiles
quartiles = percentile(data,[25,50,75])

In [4]:
#calculating min and max
data_min,data_max=data.min(),data.max()

In [5]:
#print summary
print("min:%.3f" % data_min)
print("Q1:%.3f" % quartiles[0])
print("Q1:%.3f" % quartiles[1])
print("Q1:%.3f" % quartiles[2])
print("max:%.3f" % data_max)

min:0.001
Q1:0.245
Q1:0.486
Q1:0.732
max:1.000


The result shows very similar numbers to the respective quartiles. 0.24 for the 25th percentile, .50 for the 50th percentile and .75 for the 75th percentile.